### PR TITLE
Enable emojis only if UTF-8 is the JVM default charset

### PR DESCRIPTION
**Pull request by kwart**
This PR improves enabling emojis (Hazelcast property `hazelcast.logging.emoji.enabled`).

They are not enabled if `UTF-8` is not the JVM default charset.

Resolves hazelcast/hazelcast-enterprise#4787

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java [hunk 1]
```diff
@@ -1756,7 +1758,7 @@ private int getWhenNoSSLDetected() {
      * @since 5.0
      */
     public static final HazelcastProperty LOG_EMOJI_ENABLED = new HazelcastProperty("hazelcast.logging.emoji.enabled",
-            !OsHelper.isWindows());
+            StandardCharsets.UTF_8.equals(Charset.defaultCharset()));
```
Comment (ldziedziul) [inline hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: Does it work on Windows with UTF-8 as the default charset?

Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: Even when I change the system regional settings to the UTF-8 support, `java` has the Cp1252 default charset.

image

Still, I'm no Windows expert :)
@ramizdundar Do you see any issue with the default charset-based approach?

Comment (ldziedziul) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: https://inside.java/2021/10/04/the-default-charset-jep400/

> TL;DR: Starting JDK 18, UTF-8 is the default charset across platforms. Make sure to test your applications, especially if you are running them on Windows.

Comment (ldziedziul) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: Of course, it's nothing critical, I can imagine we can fix any Windows issues as they pop out

Comment (ramizdundar) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: Here is what I tried:

1. Get a clean build with this branch on Windows 11 and Corretto 8.
2. Extract distribution.
3. Modify hz-start.bat and add this ` "-Dfile.encoding=UTF8"^` to `JAVA_OPTS`.

Here is what I observed:
```
2022-03-01 19:07:57,015 [ INFO] [main] [c.h.s.security]: [127.0.0.1]:5701 [dev] [5.2-SNAPSHOT] Enable DEBUG/FINE log level for log category com.hazelcast.system.security  or use -Dhazelcast.security.recommendations system property to see ≡ƒöÆ security recommendations and the status of current config.
```

`to see ≡ƒöÆ security`

Note that this is probably because of https://github.com/microsoft/terminal/issues/190, but not 100% sure. With IntelliJ terminal everything works fine. So I can approve it but we should note that both cmd and win terminal wont support emojis, so users will see what I see in the future. But it's possible that when we support jdk 18 win terminal probably support emojis.

**Review by ramizdundar — APPROVED** [review]

**Review by ldziedziul — APPROVED** [review]

File: hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java (hunk 1 above)
Comment (kwart) [reply hazelcast/src/main/java/com/hazelcast/spi/properties/ClusterProperty.java:1761]: I've tested different ways on Windows 10 consoles (`cmd` and `powershell`):
* `chcp 65001`
* use different fonts (Lucida console, Free Mono, Liberation Mono, DejaVu Sans Mono)
* switch to UTF-8 Beta support in the language settings

No approach lead to the successful lock display.

So given the fact, `UTF-8` is the default for new Java versions on Windows I will extend the condition to 

```
the charset is UTF-8 and the system is not Windows
```

**Review by ramizdundar — APPROVED** [review]
I also agree with this, we can always enable emojis when we think it will work by default. Thanks for the fix.

**Comment by kwart** [comment]
Thanks, guys for the reviews and comments!
